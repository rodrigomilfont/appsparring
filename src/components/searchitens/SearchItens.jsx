import React from 'react'
import PropTypes from 'prop-types';

const SearchItens = props => {
  const { searchs } = props;
  return (
    <ul>
      {searchs.map(item => (
        <li key={item.id}>
          <p>{item.title}</p>
          <img src={item.thumbnail} alt={item.title} />
          <span>
            {new Date(item.stop_time).toLocaleTimeString('pt-BR', {
              timeZone: 'UTC',
            })}
          </span>
        </li>
      ))}
    </ul>
  );
};

SearchItens.propTypes = {
  searchs: PropTypes.arrayOf(PropTypes.object),
};

SearchItens.defaultProps = {
  searchs: [],
};

export default SearchItens;
