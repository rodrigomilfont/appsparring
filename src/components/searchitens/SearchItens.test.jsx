import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import SearchItens from './SearchItens';

Enzyme.configure({ adapter: new Adapter() });

it('render Search Itens', () => {
  const searchs = [
    {
      title: 'Hello',
      id: 0,
      thumbnail: 'url',
      stop_time: '2037-12-30T04:00:00.000Z',
    },
  ];

  const wrapper = shallow(<SearchItens searchs={searchs} />);
  expect(wrapper).toMatchSnapshot();
});
