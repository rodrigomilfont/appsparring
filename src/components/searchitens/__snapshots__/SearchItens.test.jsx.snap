// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Search Itens 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchItens
    searchs={
      Array [
        Object {
          "id": 0,
          "stop_time": "2037-12-30T04:00:00.000Z",
          "thumbnail": "url",
          "title": "Hello",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <p>
            Hello
          </p>
          <img
            alt="Hello"
            src="url"
          />
          <span>
            04:00:00
          </span>
        </li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Hello
            </p>,
            <img
              alt="Hello"
              src="url"
            />,
            <span>
              04:00:00
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hello",
            },
            "ref": null,
            "rendered": "Hello",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Hello",
              "src": "url",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "04:00:00",
            },
            "ref": null,
            "rendered": "04:00:00",
            "type": "span",
          },
        ],
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <p>
              Hello
            </p>
            <img
              alt="Hello"
              src="url"
            />
            <span>
              04:00:00
            </span>
          </li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Hello
              </p>,
              <img
                alt="Hello"
                src="url"
              />,
              <span>
                04:00:00
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hello",
              },
              "ref": null,
              "rendered": "Hello",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Hello",
                "src": "url",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "04:00:00",
              },
              "ref": null,
              "rendered": "04:00:00",
              "type": "span",
            },
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
